Look into upgrading datatype-expansion dependency

Newest version of datatype-expansion is 0.1.1 but it causes one failing test

```
  32 passing (3s)
  1 failing

  1) raml2obj worldmusic.raml should test the /api methods:

      AssertionError: 'union' === 'object'
      + expected - actual

      -union
      +object
      
      at Context.it (test/worldmusic.spec.js:100:14)
```

--- a/index.js
+++ b/index.js
@@ -74,15 +74,14 @@ function _expandRootTypes(types) {
   }
 
   Object.keys(types).forEach(key => {
-    tools.expandedForm(types[key], types, (err, expanded) => {
-      if (expanded) {
-        tools.canonicalForm(expanded, (err2, canonical) => {
-          if (canonical) {
-            types[key] = canonical;
-          }
-        });
-      }
-    });
+    try {
+      const expanded = tools.expandedForm(types[key], types);
+      const canonical = tools.canonicalForm(expanded);
+      types[key] = canonical;
+    } catch (err) {
+      // Dump the error to stderr and continue with the non-canonical form
+      console.error(err);
+    }
   });
 
   return types;

--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
     "url": "https://github.com/raml2html/raml2obj/issues"
   },
   "dependencies": {
-    "datatype-expansion": "0.0.15",
+    "datatype-expansion": "0.2.4",
     "raml-1-parser": "1.1.31"
   },
   "devDependencies": {

--- a/test/worldmusic.spec.js
+++ b/test/worldmusic.spec.js
@@ -111,12 +111,41 @@ describe('raml2obj', () => {
       assert.strictEqual(post.body.length, 1);
       assert.strictEqual(post.body[0].name, 'RamlDataType');
       assert.strictEqual(post.body[0].key, 'application/json');
-      assert.strictEqual(post.body[0].type, 'object');
-      assert.strictEqual(post.body[0].properties.length, 14);
+      assert.strictEqual(post.body[0].type, 'union');
+      assert.strictEqual(post.body[0].anyOf.length, 4);
+      assert.strictEqual(post.body[0].anyOf[0].properties.length, 14);
       assert.strictEqual(
-        post.body[0].properties[4].examples[0].value,
+        post.body[0].anyOf[0].properties[4].examples[0].value,
         'very well made'
       );
+      assert.strictEqual(post.body[0].anyOf[0].properties[10].key, 'NilValue');
+      assert.strictEqual(
+        post.body[0].anyOf[0].properties[10].properties[1].type,
+        'string'
+      );
+      assert.strictEqual(post.body[0].anyOf[0].properties[11].key, 'CatOrDog');
+      assert.strictEqual(post.body[0].anyOf[0].properties[11].name, 'Cat');
+      assert.strictEqual(post.body[0].anyOf[1].properties[10].key, 'NilValue');
+      assert.strictEqual(
+        post.body[0].anyOf[1].properties[10].properties[1].type,
+        'nil'
+      );
+      assert.strictEqual(post.body[0].anyOf[1].properties[11].key, 'CatOrDog');
+      assert.strictEqual(post.body[0].anyOf[1].properties[11].name, 'Cat');
+      assert.strictEqual(post.body[0].anyOf[2].properties[10].key, 'NilValue');
+      assert.strictEqual(
+        post.body[0].anyOf[2].properties[10].properties[1].type,
+        'string'
+      );
+      assert.strictEqual(post.body[0].anyOf[2].properties[11].key, 'CatOrDog');
+      assert.strictEqual(post.body[0].anyOf[2].properties[11].name, 'Dog');
+      assert.strictEqual(post.body[0].anyOf[3].properties[10].key, 'NilValue');
+      assert.strictEqual(
+        post.body[0].anyOf[3].properties[10].properties[1].type,
+        'nil'
+      );
+      assert.strictEqual(post.body[0].anyOf[3].properties[11].key, 'CatOrDog');
+      assert.strictEqual(post.body[0].anyOf[3].properties[11].name, 'Dog');
     });
 
     it('should test the /entry resource', () => {

